### PR TITLE
Account for non-existing schedule route id for the Green Line

### PR DESCRIPTION
**Pull request by alicia-mbta**
#### Summary of changes
**Asana Ticket:** [🐞 Elixir.UndefinedFunctionError: function nil.route/0 is undefined](https://app.asana.com/0/385363666817452/1185865721068541)

Another bug involving trip info. This fix avoids getting a `500` error

### CODE REVIEW

**Review by thecristen — CHANGES_REQUESTED** [review]
Looks good on first glance! Let's not commit the formatting improvement to `VoteMap` though, since we'll be reverting that code soon.

**Comment by alicia-mbta** [comment]
That was just the linter 😄 
Relatedly, I also noticed that the build fails because there is not enough code coverage due to the new "polling location finder" work (I believe).
I think this bug fix is a "nice to have" so it probably can wait to be merged after the revert.
Let me know what you think.

**Comment by thecristen** [comment]
Gotcha. You're correct about the coverage failure reason. I agree with you on the wait - and I'll re-review then too :)

File: apps/site/lib/site_web/controllers/schedule/finder_api.ex [hunk 1]
```diff
@@ -450,14 +461,21 @@ defmodule SiteWeb.ScheduleController.FinderApi do
...
 
-    schedule.route.id
+    if Enum.empty?(schedule_for_trip) do
+      nil
+    else
+      schedule =
+        schedule_for_trip
+        |> List.first()
+
+      schedule.route.id
```
Comment (thecristen) [inline apps/site/lib/site_web/controllers/schedule/finder_api.ex:477]: I think [CodeCov wants a new test to cover this piece](https://codecov.io/gh/mbta/dotcom/src/19b90a0c42932ac61a2fcb52d53156098e008cb6/apps/site/lib/site_web/controllers/schedule/finder_api.ex), maybe that'll make it pass?

**Review by thecristen — APPROVED** [review]

**Pull request merged**